compiler/pipes/parse-and-apply-phpdoc.cpp

```diff
@@ -525,33 +526,53 @@ class ParseAndApplyPhpDocForClass {
 
 public:
   explicit ParseAndApplyPhpDocForClass(ClassPtr klass) : klass(klass) {
+    // apply @kphp-serializable and so on
+    if (klass->phpdoc) {
+      for (const PhpDocTag &tag : klass->phpdoc->tags) {
+        parse_kphp_doc_tag(tag);
+      }
+    }
 
+    std::size_t members_count{0};
     // if there is @var / type hint near class field — save it; otherwise, convert the field default value to a type hint
     // note: it's safe to use init_val here (even if it refers to constants of other classes): defines were inlined at previous pipe
     klass->members.for_each([&](ClassMemberInstanceField &f) {
+      ++members_count;
       stage::set_location(f.root->location);
       f.type_hint = calculate_field_type_hint(f.phpdoc, f.type_hint, f.var, klass);
       if (f.type_hint) {
         f.type_hint = phpdoc_finalize_type_hint_and_resolve(f.type_hint, klass->get_holder_function());
         kphp_error(f.type_hint, fmt_format("Failed to parse @var of {}", f.var->as_human_readable()));
       }
+
+      if (f.phpdoc) {
+        for (const PhpDocTag &tag : f.phpdoc->tags) {
+          if (tag.type == PhpDocType::kphp_json) {
+            f.kphp_json_tags->parse_tag(tag.value);
+          }
+        }
+        f.kphp_json_tags->complete_phpdoc_parsing();
+      }
+      f.kphp_json_tags->merge_with(*klass->kphp_json_tags, f.local_name());
     });
+
+    if (klass->kphp_json_tags->has_flatten()) {
+      kphp_error_return(members_count == 1, fmt_format("kphp-json flatten tag allowed only for class with single field, class name {}", klass->name));
```

Comment:
'@kphp-json flatten' allowed only for class with a single field, ...

Reply:
fixed